lib/models/BucketLoggingStatus.ts

```diff
@@ -0,0 +1,137 @@
+    <TargetBucket>${this._loggingEnabled.TargetBucket}</TargetBucket>
+    <TargetPrefix>${this._loggingEnabled.TargetPrefix}</TargetPrefix>
+</LoggingEnabled>
+`;
+        }
+
+        return `<?xml version="1.0" encoding="UTF-8"?>
+<BucketLoggingStatus xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
+    ${loggingEnabledXML}
+</BucketLoggingStatus>
+`;
+    }
+
+    static async fromXML(
+        data: string,
+    ): Promise<{ error?: { arsenalError: ArsenalError, details: any }; res?: BucketLoggingStatus; }> {
+        let parsed;
+        try {
+            parsed = await parseStringPromise(data);
+        } catch (err) {
+            return {
+                error: { arsenalError: errors.MalformedXML, details: err },
+                res: undefined,
+            };
+        }
+
+        if (!parsed) {
+            return {
+                error: { arsenalError: errors.MalformedXML, details: 'request xml is undefined or empty' },
+                res: undefined,
+            };
+        }
+
+        if (!parsed.BucketLoggingStatus) {
+            return {
+                error: { arsenalError: errors.MalformedXML, details: 'missing BucketLoggingStatus root tag' },
+                res: undefined,
+            };
+        }
+
+        let loggingEnabled : LoggingEnabled | undefined = undefined; 
+        if (parsed.BucketLoggingStatus.LoggingEnabled) {
+            if (!parsed.BucketLoggingStatus.LoggingEnabled.TargetBucket) {
+                return {
+                    error: { arsenalError: errors.MalformedXML, details: 'missing TargetBucket tag in LoggingEnabled' },
+                    res: undefined,
+                };
+            }
+
+            if (!parsed.BucketLoggingStatus.LoggingEnabled.TargetPrefix) {
+                return {
+                    error: { arsenalError: errors.MalformedXML, details: 'missing TargetPrefix tag in LoggingEnabled' },
+                    res: undefined,
+                };
+            }
+
+            if (parsed.BuckerLoggingStatus.LoggingEnabled.TargetGrants) {
+                return {
+                    error: {
+                        arsenalError: errors.NotImplemented,
+                        details: 'TargetGrants tag in LoggingEnabled is not implemented',
+                    },
+                    res: undefined,
+                };
+            }
+
+            if (parsed.BuckerLoggingStatus.LoggingEnabled.TargetObjectKeyFormat) {
+                return {
+                    error: {
+                        arsenalError: errors.NotImplemented,
+                        details: 'TargetObjectKeyFormat tag in LoggingEnabled is not implemented',
+                    },
+                    res: undefined,
+                };
+            }
+
+            loggingEnabled = {
+                TargetPrefix: parsed.BucketLoggingStatus.LoggingEnabled.TargetPrefix[0],
+                TargetBucket: parsed.BucketLoggingStatus.LoggingEnabled.TargetBucket[0],
```

Comment:
Aws performs no validation on those fields ?

Because bucket name is pretty restrictive format, and prefix should probably not go beyond a certains size